### ISSUE TITLE
Getting: Error: underlying network changed 

### ISSUE THREAD

**Issue by FabienCoutant** [NoahZinsmeister/web3-react#582]
Hi, I'm using web3-react v8 in my app but getting the following error on network change using a select network or directly from metamask:

```
index.js?4a96:219 Uncaught (in promise) Error: underlying network changed (event="changed", network={"name":"matic","chainId":137,"ensAddress":null}, detectedNetwork={"name":"homestead","chainId":1,"ensAddress":"0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e"}, code=NETWORK_ERROR, version=providers/5.6.8)
```
I'm calling a getBalances() function in order to fetch the user's balances of multi tokens and need to pass the web3-react provider to a multicall function. Bellow my hooks called from my components:

```
export const useGetUserAssetsBalances = (assets: any): TokenWithBalance[] => {
  const { account, provider } = useWeb3React();
  const [balances, setBalances] = useState<TokenWithBalance[]>([]);
  const fetchBalances = useCallback(async () => {
    const tokens: Token[] = Object.values(assets);
    let data: TokenWithBalance[] = [];
    if (account && provider) {
      data = (await getMultipleTokenBalancesByMultiCall(
        tokens,
        account,
        provider,
      )) as TokenWithBalance[];
    }
    setBalances(data);
  }, [account, provider, assets]);

  useEffect(() => {
    if (account && assets && provider && chainId) {
      fetchBalances();
    }
  }, [account, provider, assets, setBalances, fetchBalances]);
  return balances;
};
```
The `getMultipleTokenBalancesByMultiCall()` is defined as below : 

```
export const getMultipleTokenBalancesByMultiCall = async (
  tokens: Token[],
  userAddress: string,
  provider: ethers.providers.Provider
): Promise<TokenWithBalance[]> => {
  const ethCallProvider = new Provider(provider);
  await ethCallProvider.init();
  const balanceCalls = tokens.map((token) => {
    if (NATIVE[chainId].equals(token)) {
      return ethCallProvider.getEthBalance(userAddress);
    }
    return getERC20Contract_MultiCall(token.address).balanceOf(userAddress);
  });
  const balances = await ethCallProvider?.all(balanceCalls);
  return tokens.map(
    (token, index) =>
      ({
        ...token,
        balance: FixBigNumber.fromWei(
          balances[index].toString(),
          token.decimals
        ),
      } as TokenWithBalance)
  );
};
```

I tried to remove the `getMultipleTokenBalancesByMultiCall()` and just make a `await provider?.getNetwork();`, but getting the same error. 

I'm using `import { useWeb3React } from '@web3-react/core';` 

My web3react dependencies are: 
```
    "@coinbase/wallet-sdk": "^3.1.0",
    "@walletconnect/ethereum-provider": "^1.7.8",
    "@web3-react/coinbase-wallet": "8.0.31-beta.0",
    "@web3-react/core": "8.0.30-beta.0",
    "@web3-react/frame": "8.0.17-beta.0",
    "@web3-react/metamask": "8.0.24-beta.0",
    "@web3-react/network": "8.0.23-beta.0",
    "@web3-react/types": "8.0.17-beta.0",
    "@web3-react/walletconnect": "8.0.31-beta.0",
```
 and using the `import { Web3ReactProvider } from '@web3-react/core';` that wrap every components in my _app.tsx file

I don't have any idea on why my provider coming from useWeb3React is not sync with the current network. Do someone has an idea?

**Comment by tonisives** [comment]
I had this error when I input

```
['http://127.0.0.1:8545']
```

to the `initializeConnector<Network>`.

The input should actually be of type `[chainId: number]: string[]`, eg

```
{
	[31337]: ['http://127.0.0.1:8545']
}
```

There should be a type error if you input wrong type to the connector ideally.

**Comment by FabienCoutant** [comment]
My error has been fixed version 8.0.31-beta.0 of @web3-react/core
